docs/06-appearance.md.lua

```diff
@@ -0,0 +1,222 @@
+#! /usr/bin/lua
+local args = {...}
+
+local gio = require("lgi").Gio
+local gobject = require("lgi").GObject
+local glib = require("lgi").GLib
+
+local name_attr = gio.FILE_ATTRIBUTE_STANDARD_NAME
+
+-- Recursive file scanner
+local function get_all_files(path, ext, ret)
+    ret = ret or {}
+    local enumerator = gio.File.new_for_path(path):enumerate_children(
+        "FILE_ATTRIBUTE_STANDARD_NAME", 0, nil, nil
+    )
+
+    for file in function() return enumerator:next_file() end do
+        local file_name = file:get_attribute_as_string(name_attr)
+        local file_type = file:get_file_type()
```

Comment:
Please look at `lib/menubar/utils.lua`, function `utils.parse_dir`. You have to explicitly request the file type, since otherwise some filesystems (XFS?) do not provide it. Also, instead of giving a string to `enumerate_children`, I would prefer the approach used in menubar since it catches typos (it still provides a string argument, but the string is not hardcoded but comes from gio).